### PR TITLE
Support OAuth authentication for the Bigcommerce API.

### PR DESCRIPTION
**Pull request by pedelman**
This code is pulled in the mechatama/bigcommerce-api-ruby repo on the
oauth branch.

- Updates to the README
- OAuth support
- Bump minor version
- Remove unnecessary code

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 require 'bundler/gem_tasks'
-require 'ci/reporter/rake/rspec'
+#require 'ci/reporter/rake/rspec'
```
Comment (mattolson) [inline Rakefile]: Please add the missing gem or remove this altogether.

**Comment by mattolson** [comment]
Overall, looks good, thanks for this. Please fix the ci-reporter stuff.

File: Rakefile [hunk 2]
```diff
@@ -9,3 +9,4 @@ RSpec::Core::RakeTask.new(:spec) do |spec|
 end
 
 task :default => ['ci:setup:rspec', :spec]
+task :default => [:spec]
```
Comment (mattolson) [inline Rakefile]: Why two default tasks?

Comment (pedelman) [inline Rakefile]: My bad :)

**Comment by mattolson** [comment]
Awesome, thank you!

**Pull request merged**